pandas/tests/indexes/test_interval.py

```diff
@@ -657,206 +635,39 @@ def testcontains(self):
 
-        tm.assert_numpy_array_equal(idx.covers(Interval(1, 3, closed='both')), np.array([1, 2]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(0, 5, closed='both')), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(0, 2, closed='both')), np.array([0]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(2, 4, closed='both')), np.array([1]))
-
-    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
-    def test_intervalIndex_covers_intervalIndex(self):
-
-        # class IntervalIndex:
-        #     def covers(self, other: IntervalIndex) -> Tuple[IntegerArray1D, IntegerArray1D]
-
-        idx1 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
-        idx2 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='left')
-        idx3 = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='both')
-
-        self._compare_tuple_of_numpy_array(idx.covers(idx1), (np.array([0,1,2,2]), np.array([0,1,1,2])))
-        self._compare_tuple_of_numpy_array(idx.covers(idx2), (np.array([2]), np.array([1])))
-        self._compare_tuple_of_numpy_array(idx.covers(idx3), (np.array([0,1,2,2]), np.array([0,1,1,2])))
-
-    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
-    def test_intervalIndex_overlaps_interval_updated_behavior(self):
-
-        # class IntervalIndex:
-        #     def overlaps(self, other: Interval) -> IntegerArray1D
-
-        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
-
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 3, closed='right')), np.array([1, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 2, closed='right')), np.array([2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(0, 2, closed='right')), np.array([0, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(3, 4, closed='right')), np.array([]))
-
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 3, closed='left')), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 2, closed='left')), np.array([0, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(0, 2, closed='left')), np.array([0, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(3, 4, closed='left')), np.array([3]))
-
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 3, closed='both')), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(1, 2, closed='both')), np.array([0, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(0, 2, closed='both')), np.array([0, 2]))
-        tm.assert_numpy_array_equal(idx.overlaps(Interval(3, 4, closed='both')), np.array([3]))
-
-    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
-    def test_intervalIndex_overlaps_intervalIndex_updated_behavior(self):
+        assert not index.contains(0)
+        assert index.contains(0.1)
```

Comment:
Can you do the `in` tests and `contains` tests in separate test functions? (as IMO they should have distinct behaviour)